sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample00_AuthenticateClient.md

```diff
@@ -0,0 +1,56 @@
+
+```C# Snippet:ServiceBusAuthConnString
+// Create a ServiceBusClient that will authenticate using a connection string
+string connectionString = "<connection_string>";
+await using var client = new ServiceBusClient(connectionString);
+```
+
+## Authenticate with Azure.Identity
+The [Azure Identity library](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity/README.md) provides [role-based access control](https://docs.microsoft.com/azure/role-based-access-control/overview) support for authentication using Azure Active Directory.In order to leverage role-based access control for Service Bus, please refer to the [role-based access control documentation](https://docs.microsoft.com/azure/service-bus-messaging/service-bus-role-based-access-control). The simplest way to get started using the `Azure.Identity` `library is to use the [DefaultAzureCredential](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet).
+
+```C# Snippet:ServiceBusAuthAAD
+// Create a ServiceBusClient that will authenticate through Active Directory
+string fullyQualifiedNamespace = "yournamespace.servicebus.windows.net";
+await using var client = new ServiceBusClient(fullyQualifiedNamespace, new DefaultAzureCredential());
+```
+
+## Authenticate with a Shared Access Key Credential
+
+Shared access keys for Service Bus authorization are generated when access policies are created for a Service Bus namespace. Since these keys are most often used in association with a connection string, the article providing instructions to [get a Service Bus connection string](https://docs.microsoft.com/azure/service-bus-messaging/service-bus-quickstart-topics-subscriptions-portal#get-the-connection-string) is the best source of information on generating and accessing them. One of the key benefits to using a shared access key credential over a connection string is that the credential allows support for updating the key without creating a new client. In step 3 of the article, the policy that you select will be the name of your shared access key when used for credential authorization. You'll want to copy the "Primary key" rather than connection string.
+
+
+```C# Snippet:ServiceBusAuthNamedKey
+var credential = new AzureNamedKeyCredential("<name>", "<key>");
+await using var client = new ServiceBusClient("yournamespace.servicebus.windows.net", credential);
+```
+
+## Authenticate with a Shared Access Signature Credential
```

Comment:
This is a good idea as well.    I don't think the `url` below is correct.   Shouldn't it be more like:

```csharp
private static string BuildAudience(
    string fullyQualifiedNamespace,
    string entityName)
{    
    var builder = new UriBuilder(fullyQualifiedNamespace)
    {
        Scheme = "amqps",
        Path = entityName,
        Port = -1,
        Fragment = string.Empty,
        Password = string.Empty,
        UserName = string.Empty,
    };

    builder.Path = builder.Path.TrimEnd('/');
    return builder.Uri.AbsoluteUri.ToLowerInvariant();
}
```

By the time we get a namespace name, I think we strip the protocol and such out.

Reply:
You are correct that the protocol is not there, but it works without the protocol, so I guess it is not necessary.

Reply:
Interesting.  My understanding was that it needed to match exactly.    That said, without the entity name, you're demonstrating a namespace-level grant, which goes against the principle of least privilege.

Reply:
I'm good if you decide that you want to keep it simple, but please throw a big disclaimer if you choose to do so and mention that they should refer to the SAS guide that you linked for demonstration of creating one scoped to the entity.

Reply:
I tested it out and the scheme seems to not matter at all - you can specify any thing or leave it off entirely. Agreed on including entity path.